client/signup/steps/website-content/index.tsx

```diff
@@ -39,13 +45,43 @@ function WebsiteContentStep( {
 	goToNextStep,
 	queryObject,
 }: WebsiteContentStepProps ) {
+	const [ formErrors, setFormErrors ] = useState< ValidationErrors >( {} );
+	const dispatch = useDispatch();
+	const translate = useTranslate();
+
 	const websiteContent = useSelector( getWebsiteContent );
 	const currentIndex = useSelector( getWebsiteContentDataCollectionIndex );
 	const siteId = useSelector( ( state ) => getSiteId( state, queryObject.siteSlug as string ) );
 	const siteCategory = useSelector( ( state ) => getDIFMLiteSiteCategory( state, siteId ) );
-	const dispatch = useDispatch();
-	const translate = useTranslate();
-	const [ formErrors, setFormErrors ] = useState< ValidationErrors >( {} );
+	const isWebsiteContentSubmitted = useSelector( ( state ) =>
+		isDIFMLiteWebsiteContentSubmitted( state, siteId )
+	);
+	const isSiteInformationLoaded = useSelector( ( state ) =>
+		isRequestingSite( state, siteId as number )
+	);
+
+	// We assume that difm lite is purchased when the is_difm_lite_in_progress sticker is active in a given blog
+	const isDifmLitePurchased = useSelector( ( state ) => isDIFMLiteInProgress( state, siteId ) );
+
+	//Make sure site information is loaded so that we can validate access to this page
+	useEffect( () => {
+		siteId && dispatch( requestSite( siteId ) );
```

Comment:
I was wondering if this is required since the signup controller already makes a request to fetch the site. You shared the relevant code here: https://github.com/Automattic/wp-calypso/pull/61121#issuecomment-1041402184


https://github.com/Automattic/wp-calypso/blob/4dd521885f9d795412714a6ece6f4315d6dbb7b5/client/signup/controller.js#L407-L412

Reply:
Ah yup I will test without the dispatch :D

Reply:
In case the site information was loaded before checkout the site information now needs to be refreshed for the options to be updated. Do you have a better way in mind to do this?